controllers/storagecluster/storagecluster_controller.go

```diff
@@ -129,6 +130,21 @@ func (r *StorageClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		},
 	}
 
+	ocsClientOperatorStatusPredicate := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			if e.ObjectOld == nil || e.ObjectNew == nil {
+				return false
+			}
+			oldObj := e.ObjectOld.(*ocsv1alpha1.StorageConsumer)
+			// Nullifying heartbeat field in order to compare the status correctly
+			initTime := metav1.Time{}
+			oldObj.Status.LastHeartbeat = initTime
+			newObj := e.ObjectNew.(*ocsv1alpha1.StorageConsumer)
+			newObj.Status.LastHeartbeat = initTime
+			return reflect.DeepEqual(oldObj.Status, newObj.Status)
```

Comment:
Although I understand the intention here, the methodology is very very fragile. You cannot assume that we will not add new volatile fields to the `StorageConsumer` status (similar to  `LastHeartbeat`). If you are interested in changes to a known subset of the consumer status, you should compare the values of these specific fields included in that subset, not the entire status.

Reply:
I'm not fully convinced. The assumption I made was that volatile fields are less common and so it makes more sense to go with the approach of adding to the exclusion list.

Reply:
Think about it differently, there is a subset of the status that is of interest for your PR. That subset is fixed and known at this point in time. If that subset of things that are important for the CM will grow then someone will need to write code to handle the new stuff (or else it is not important for the CM). 

With that in mind, let's consider white list vs black list:

Using a white list you can guarantee that you are reacting on the exact changes that are needed for CM generation. No more, no less. If someone adds anything to the status that is not relevant for the CM generation you will not react on it. If someone adds a new field that is important for the CM generation then they will need to write code to handle the change in the generation code and can update the white list at the same time. 
This makes sure that we are always fully and exactly aligned with the needs of this controller!

On the other hand, using a blacklist means that if I am working on something unrelated to CM generation and add a new field to the status, your CM logic will get unnecessarily invoked on any change to that new field. And as someone who works on an unrelated feature, I will not even think to add anything to any black list. 

This is exactly why I think a white list is the better approach here

Reply:
Okay, thanks for the thorough explanation.